### PR TITLE
DAOS-9881 doc: improve perf tuning section

### PR DESCRIPTION
**Pull request by johannlombardi**
Add more examples for self_test and always use the agent.
Documment spdk_nvme_perf.
Fix nit which is a source of confusion in dual-rail section.

Signed-off-by: Johann Lombardi <johann.lombardi@intel.com>

### CODE REVIEW

**Review by wangshilong — DISMISSED** [review]
Thanks for fix net parts. *_*

File: docs/admin/performance_tuning.md [hunk 1]
```diff
@@ -1,229 +1,165 @@
...
-```bash
-$ git clone --recurse-submodules https://github.com/daos-stack/daos.git
-$ cd daos
-$ scons --build-deps=yes install
-$ cd install
-```
+- tag 0 is used by the metadata service handling pool and container operations.
+- tag 1 is used for cross-server monitoring (SWIM).
+- tags 2 to [#targets + 1] is used by DAOS targets (one tag per target].
+- tags [#targets + 1] to [#targets + #helpers] is used by helper service threads.
```
Comment (Michael-Hennecke) [inline docs/admin/performance_tuning.md:42]: add 2 to the ranges

File: docs/admin/performance_tuning.md [hunk 2]
```diff
@@ -1,229 +1,165 @@
 # DAOS Performance Tuning
 
+This section aims at documenting how to validate the performance of the baseline
```
Comment (Michael-Hennecke) [inline docs/admin/performance_tuning.md:3]: "aims at documenting" --> "documents"

File: docs/admin/performance_tuning.md [hunk 3]
```diff
@@ -1,229 +1,165 @@
...
 
 The DAOS [CaRT][1] layer can validate and benchmark network communications in
 the same context as an application and using the same networks/tuning options
 as regular DAOS.
 
 The CaRT `self_test` can run against the DAOS servers in a production environment
-in a non-destructive manner. CaRT `self_test` supports different message sizes,
-bulk transfers, multiple targets, and the following test scenarios:
+in a non-destructive manner. The only requirement is to have the DAOS agent
+running on the client node where self\_test is run.
```
Comment (Michael-Hennecke) [inline docs/admin/performance_tuning.md:15]: and daos_server running on the server(s)?  Does the storage need to be formatted?

File: docs/admin/performance_tuning.md [hunk 4]
```diff
@@ -1,229 +1,165 @@
...
+- tag 0 is used by the metadata service handling pool and container operations.
+- tag 1 is used for cross-server monitoring (SWIM).
+- tags 2 to [#targets + 1] is used by DAOS targets (one tag per target].
+- tags [#targets + 1] to [#targets + #helpers] is used by helper service threads.
 
-For detailed information, please refer to the [DAOS build documentation][3]
-section.
+As an example, an engine with `targets: 16` and `nr_xs_helpers: 4` would use the
+following tag distributions:
+- tag 0: metadata service
```
Comment (Michael-Hennecke) [inline docs/admin/performance_tuning.md:46]: blank line before bullet list?

File: docs/admin/performance_tuning.md [hunk 5]
```diff
@@ -1,229 +1,165 @@
...
+For a total of 21 endpoints exported by this engine.
 
-Instructions to run CaRT `self_test` are as follows.
+The RPC flow sent over the network can be configured via the `--message-sizes`
+options that take a list of size tuples. Performance will be reported individually
+for each tuple. Each size integer can be prepended with a single character to specify
+the underlying transport mechanism. Available types are:
 
-**Start DAOS server**
+- 'e' - Empty (no payload)
```
Comment (Michael-Hennecke) [inline docs/admin/performance_tuning.md:58]: blank line?

File: docs/admin/performance_tuning.md [hunk 6]
```diff
@@ -1,229 +1,165 @@
...
+determines the number of concurrent RPCs issued simultaneously.
 
-`self_test` will use the address information in `daos_server.attach_info_tmp`
-file. To create such file, run the following command:
+`self_test` has many more options. For a full description of `self_test` usage,
+please run:
 
 ```bash
-./bin/daos_agent dump-attachinfo -o ./daos_server.attach_info_tmp
+$ ./bin/self_test --help
```
Comment (Michael-Hennecke) [inline docs/admin/performance_tuning.md:73]: drop "./bin/" ? should be in the PATh for ROM install, and we don't say what ".". is...

File: docs/admin/performance_tuning.md [hunk 7]
```diff
@@ -1,229 +1,165 @@
...
+### Example: Client-to-Servers
 
-The example below uses an Ethernet interface and TCP provider.
-In the `self_test` commands:
+To run self\_test in client-to-servers mode:
 
--   **Selftest client to servers** - Replace the argument for `--endpoint`
-    accordingly.
+```bash
+self_test -u --group-name daos_server --endpoint 0:2 --message-size '(0 b1048578)' --max-inflight-rpcs 16 --repetitions 100000
```
Comment (Michael-Hennecke) [inline docs/admin/performance_tuning.md:81]: drop "--group-name daos_server" as this is the default?   same below

File: docs/admin/performance_tuning.md [hunk 8]
```diff
@@ -1,229 +1,165 @@
...
+    This can be forced by setting the OFI\_INTERFACE and OFI\_DOMAIN environment
+    variables manually. e.g. export OFI\_INTERFACE=eth0; export OFI\_DOMAIN=eth0
+    or export OFI\_INTERFACE=ib0; export OFI\_DOMAIN=mlx5_0
 
-The IO-500 workloads for mdtest provide some good criteria for performance
-measurements.
+!!! note
+    Depending on the HW configuration, the agent might assign a different
+    network interface to the self_test application depending on the NUMA node
+    where the process is scheduled. It is thus recommended to use tasket to bind
```
Comment (Michael-Hennecke) [inline docs/admin/performance_tuning.md:126]: tasket -> taskset

File: docs/admin/performance_tuning.md [hunk 9]
```diff
@@ -398,11 +334,153 @@ Completed test=FETCH
...
+spdk_nvme_perf -q 16 -o 1048576 -w read -c 0xff -t 60
+```
+
+IOPS can be measured with the following command:
+```bash
+spdk_nvme_perf -q 16 -o 4096 -w read -c 0xff -t 60
+```
+
+`-q` is used to control the queue depth, `-o` for the I/O size, `-w` is the
+operation and can be either (rand)read, (rand)write or (rand)rw. The test
```
Comment (Michael-Hennecke) [inline docs/admin/performance_tuning.md:356]: !!!note : *write and *rw are destructive

File: docs/admin/performance_tuning.md [hunk 10]
```diff
@@ -398,11 +334,153 @@ Completed test=FETCH
...
+
+This command uses all the available SSDs. Specific SSDs can be specified via the
+`--allowed-pci-addr` options followed by the PCIe addresses of the SSDs of
+interest.
+
+The `-c` option is used to specify the CPU cores used to submit I/O under the
+form of a core mash. `-c 0xff` uses the first 8 cores.
+
+!!! note
+    One storage node using Intel VMD, the `--enable-vmd` must be specified.
```
Comment (Michael-Hennecke) [inline docs/admin/performance_tuning.md:367]: must -> option must

**Review by Michael-Hennecke — CHANGES_REQUESTED** [review]
a few minor glitches, see comments

File: docs/admin/performance_tuning.md (hunk 3 above)
Comment (johannlombardi) [reply docs/admin/performance_tuning.md:15]: fixed.

File: docs/admin/performance_tuning.md (hunk 2 above)
Comment (johannlombardi) [reply docs/admin/performance_tuning.md:3]: fixed.

File: docs/admin/performance_tuning.md (hunk 1 above)
Comment (johannlombardi) [reply docs/admin/performance_tuning.md:42]: fixed.

File: docs/admin/performance_tuning.md (hunk 4 above)
Comment (johannlombardi) [reply docs/admin/performance_tuning.md:46]: there is already one actually.

File: docs/admin/performance_tuning.md (hunk 5 above)
Comment (johannlombardi) [reply docs/admin/performance_tuning.md:58]: There is already one actual.

File: docs/admin/performance_tuning.md (hunk 6 above)
Comment (johannlombardi) [reply docs/admin/performance_tuning.md:73]: fixed.

File: docs/admin/performance_tuning.md (hunk 8 above)
Comment (johannlombardi) [reply docs/admin/performance_tuning.md:126]: fixed.

File: docs/admin/performance_tuning.md (hunk 9 above)
Comment (johannlombardi) [reply docs/admin/performance_tuning.md:356]: fixed.

**Review by Michael-Hennecke — APPROVED** [review]

**Pull request merged**